app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+    alive: list["Animal"] = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, target: "Animal") -> None:
+        if not isinstance(target, Herbivore):
+            return
```

Comment:
The `bite` method only allows `Carnivore` to bite `Herbivore` instances. If this is not the intended behavior, you might want to adjust the condition to allow biting other `Animal` types as well.